twpca/model.py

```diff
@@ -381,41 +367,52 @@ def predict(self, X=None):
                      low-dimensional model prediction.
         """
         if self._sess is None:
-            raise ValueError('No model has been fit - must call TWPCA.fit() before TWPCA.predict().')
+            raise ValueError('No model has been fit - must call fit before predict.')
 
         if X is None:
             return self._sess.run(self._pred)
 
         elif isinstance(X, np.ndarray):
             # input is a (trial x time x neuron) dataset of unwarped data
             n_trials, n_timepoints, n_neurons = X.shape
+
             # grab the warped temporal factors
             if self.fit_trial_factors:
                 warped_factors = self._sess.run(self._warped_time_factors)
                 trial_factors = self._sess.run(self._params['trial'])
                 warped_factors *= trial_factors[:, None, :]     # broadcast multiply across trials
             else:
                 warped_factors = self._sess.run(self._warped_time_factors)
+
             # check input size
             if warped_factors.shape[0] != n_trials:
                 raise ValueError('Data does not have the expected number of trials.')
             if warped_factors.shape[1] != n_timepoints:
                 raise ValueError('Data does not have the expected number of timepoints.')
+
             # reshape the factors and data into matrices
             # time factors is (trial-time x components); X_unf is (trial-time x neurons)
             time_factors = warped_factors.reshape(-1, self.n_components)
             X_unf = X.reshape(-1, n_neurons)
+
             # mask nan values (only evaluate when all neurons are recorded)
             mask = np.all(np.isfinite(X_unf), axis=-1)
+
             # do a least-squares solve to fit the neuron factors
             neuron_factors = np.linalg.lstsq(time_factors[mask, :], X_unf[mask, :])[0]
+
             # reconstruct and reshape the predicted activity
             pred = np.dot(neuron_factors.T, time_factors.T)     # (neurons x trials-time)
             pred = pred.T.reshape(*X.shape)     # (trials x time x neuron)
 
             return pred
 
     @property
+    def params(self):
+        """Returns a dictionary of factors and warps"""
+        return self._sess.run(self._params)
```

Comment:
I'm not sure if there's an easy way to do this, but it would be nice if this could be smart about caching the params. For example, I have code that doesn't cache the output of `params` and just treats it like a static dict: `model.params['foo'] .... model.params['bar']`